bin/add_hypothetical_protein_descriptions.py

```diff
@@ -0,0 +1,714 @@
+            highest_match[db], "ipr_desc", "sig_desc"
+        )
+        source_description = format_source(db, source)
+        return func_description, source_description
+    else:
+        return "", ""
+
+
+def get_superfamily_info(my_dict):
+    if len(my_dict.keys()) == 1:
+        db = next(iter(my_dict.keys()))
+        description, source = pull_out_description(my_dict[db], "ipr_desc", "sig_desc")
+        source_description = format_source(db, source)
+    else:
+        db = find_higher_match(my_dict)
+        description, source = pull_out_description(my_dict[db], "ipr_desc", "sig_desc")
+        source_description = format_source(db, source)
+    return description, source_description
+
+
+def format_source(db, source):
+    if source == "ipr_desc":
+        return "InterPro({})".format(db)
+    else:
+        return db
+
+
+def get_best_match(ipr_dict):
+    best_fraction = 0
+    best_level = 0
+    highest_match = dict()
+    for db in ipr_dict:
+        if ipr_dict[db]["level"] is None:
+            ipr_dict[db]["level"] = 0
+        if not db.lower() in ["superfamily", "gene3d"] and (
+            ipr_dict[db]["sig_desc"] != "-" or ipr_dict[db]["ipr_desc"] != "-"
+        ):
+            if ipr_dict[db]["level"] > best_level or (
+                ipr_dict[db]["level"] == best_level
+                and ipr_dict[db]["match"] > best_fraction
+            ):
+                best_fraction = ipr_dict[db]["match"]
+                best_level = ipr_dict[db]["level"]
+                highest_match = dict()
+                highest_match[db] = ipr_dict[db]
+    if "Pfam" not in highest_match and best_fraction > 0.30:
+        if (
+            "Pfam" in ipr_dict
+            and best_fraction - ipr_dict["Pfam"]["match"] < 0.10
+            and ipr_dict["Pfam"]["level"] >= best_level
+        ):
+            return {"Pfam": ipr_dict["Pfam"]}
+    return highest_match
+
+
+def find_higher_match(my_dict):
+    keys = list(my_dict.keys())
+    if len(keys) != 2:
+        raise ValueError("The superfamily dictionary should have exactly 2 keys.")
+
+    key1, key2 = keys
+    match1 = my_dict[key1]["match"]
+    match2 = my_dict[key2]["match"]
+
+    if match1 > match2:
+        return key1
+    elif match2 > match1:
+        return key2
+    else:
+        return "Gene3D"  # Return Gene3d if match values are equal
+
+
+def pull_out_description(my_dict, first_priority, second_priority):
+    if not my_dict[first_priority] == "-":
+        return my_dict[first_priority], first_priority
+    else:
+        return my_dict[second_priority], second_priority
+
+
+def load_eggnog(file):
+    eggnog_info = dict()
+    with open(file, "r") as file_in:
+        for line in file_in:
+            if not line.startswith("#"):
+                cols = line.strip().split("\t")
+                try:
+                    evalue = float(cols[2])
+                except:
+                    continue
+                if evalue > 1e-10:
```

Comment:
I would also push threshold as a constant to the top of the file (or the method)